11_notify/willshen8/pkg/puppy/syncstore.go

```diff
@@ -0,0 +1,75 @@
+package puppy
+
+import (
+	"strconv"
+	"sync"
+)
+
+type SyncStore struct {
+	syncStore sync.Map
+	nextID    uint32
+	mu        sync.Mutex
+}
+
+// NewMapStore initialise a new SyncStore
+func NewSyncStore() *SyncStore {
+	return &SyncStore{syncStore: sync.Map{}}
+}
+
+// CreatePuppy create a new puppy and store in mapStore.
+func (s *SyncStore) CreatePuppy(p *Puppy) (uint32, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	i, err := strconv.Atoi(p.Value)
+	if err != nil {
+		return 0, Errorf(ErrInvalidInput, "Puppy value can't be negative")
+	}
+	if i < 0 {
+		return 0, Errorf(ErrInvalidInput, "Unrecognised input")
+	}
+	s.nextID++
```

Comment:
Same comment on `nextID` that I made on `MapStore` - this is not the next id if you increment it before using it.

Reply:
Corrected also, thanks.